src/exchange/exchange.py

```diff
@@ -337,6 +319,19 @@ def prepend_checkpointed_message(self, message: Message, token_count: int) -> No
         )
         self.checkpoint_data.message_index_offset = new_index
 
+    def rewind_to_last_user_message(self) -> None:
```

Comment:
nit: rewind_to sounds like the last user message is included, maybe `rewind_past_`

Reply:
👍🏼